src/predict_request_validation_utils.cpp

```diff
@@ -342,6 +411,31 @@ Status RequestValidator<KFSRequest, KFSTensorInputProto, KFSInputTensorIteratorT
     }
     return StatusCode::OK;
 }
+template <>
+Status RequestValidator<ovms::InferenceRequest, ovms::InferenceTensor, const ovms::InferenceTensor*, shape_t>::checkBinaryBatchSizeMismatch(const InferenceTensor& proto, const Dimension& servableBatchSize, Status& finalStatus, Mode batchingMode, Mode shapeMode) const {
+    RequestShapeInfo<InferenceTensor, shape_t> rsi(proto);
+    if (proto.getBuffer()->getByteSize() <= 0) {
```

Comment:
The error message state that the batch size is invalid - negative. You do not check batch size here but byte size, I guess thats typo?

Reply:
If thats a typo then it looks like we do not have unit test for that, we probably should add it

Reply:
Fixed.

Reply:
Cannot set negative value for u int.

Reply:
Tests can be added with binary input support.